### ISSUE TITLE
Help setting this up

### ISSUE THREAD

**Issue by FranciRoosters** [Jason-Morcos/homebridge-magic-occupancy#34]
Hello, I've been trying to set this plugin up, but I haven't been getting the desired results.

Here's what I'm trying to do:

When motion is detected in bathroom > occupancy switch 1 turns on
When occupancy switch 1 turns on > lights turn on
When no motion is detected in bathroom > occupancy switch 1 turns off, but occupancy is still detected for 60 seconds and when they're over, lights turn off.

With this first set of automations, occupancy switch 1 turns on, but it won't stay on and immediately turns off, the occupancy sensor isn't triggered and the lights turn off immediately when motion isn't detected.

When shower door opens > occupancy switch 2 turns on and shower lights stay on as long as occupancy switch 2 is on
When shower door opens > occupancy switch 2 turns off, occupancy switch 1 turns on and bathroom lights turn on
When motion is detected in bathroom > occupancy switch 1 turns on 
When no motion is detected in bathroom > occupancy switch 1 turns off, but occupancy is still detected for 60 seconds and when they're over, lights turn off.

With this second set of automations, occupancy switch 2 turns on, it stays on, but when I leave the shower I obviously run into the  the same issue described above.

The plugin is set on 60 secs right now and both switches are under "Names of Trigger Stateless Stay-on Switches". What am I doing wrong? Thanks!

**Comment by Jason-Morcos** [comment]
I think you're using the wrong types of switches here! Can you share a screenshot of the plugin settings in homebridge?

I think you don't want to be using trigger/stateless switches, you want to be using stateful switches. These trigger switches are for momentary events (ex. garage door opened) but not for long-running events (like motion happening, or lights being on). For a motion sensor, I would use a stateful switch (normal switch) and have it get flipped on when motion starts and off when motion ends. Same thing for a light too 

Additionally, stay-on switches are not able to start occupancy. Stay-on switches can only _prolong_ occupancy that has already been initiated. If you want a switch to be able to start occupancy, that's a normal switch (not a stay-on).

Once again, if you can post some screenshots of the plugin settings I'm happy to try to provide some more info. My setups are fairly similar to yours

**Comment by Jason-Morcos** [comment]
Additionally, if you haven't yet, I encourage you to fully read through the readme - near the bottom is an example that's fairly similar to what you're aiming to do

**Comment by krazyflipj** [comment]
I’m also having a hard time figuring out the plugin. What I’m trying to do is have my garage light automatically turn on when either someone opens the garage door (i have a door sensor) or if my motion sensor detects motion and for the lights to turn off once no one is in the garage. I should only be using Stateful switch?

**Comment by Jason-Morcos** [comment]
> I’m also having a hard time figuring out the plugin. What I’m trying to do is have my garage light automatically turn on when either someone opens the garage door (i have a door sensor) or if my motion sensor detects motion and for the lights to turn off once no one is in the garage. I should only be using Stateful switch?

Two switches
One for the door sensor and one for the motion sensor.
For motion sensor, definitely stateful (motion starts turn on, motion stops turn off).
For the garage door sensor if you want the lights to NOT stay on if the garage door is left open, trigger switch (on when door opened). If you want the lights to stay on if the door is left opened, stateful switch (on when opened, off when closed).

**Comment by krazyflipj** [comment]
Ok let me know if i did this right. I set up 4 automations. If my garage sensor detects motion, than the stateful switch turns on. If it doesn’t detect motion, then the stateful switch turns off. i also set up where if the occupancy sensor (from the plugin) “detects motion” my garage lights turn on and turns off when the plugin sensor doesn’t detect motion.

**Comment by Jason-Morcos** [comment]
> Ok let me know if i did this right. I set up 4 automations. If my garage sensor detects motion, than the stateful switch turns on. If it doesn’t detect motion, then the stateful switch turns off. i also set up where if the occupancy sensor (from the plugin) “detects motion” my garage lights turn on and turns off when the plugin sensor doesn’t detect motion.

Sounds right to me, sounds like that's just hooking up the motion sensor and not the garage opening part, but sounds right about motion/occupancy parts.

**Comment by Jason-Morcos** [comment]
As a note to everyone on this thread:
This plugin is really just something I made for my own automations at home. I've shared it with other users as a curtesy, but I don't provide lots of support to help other users get set up and use it (simply because there aren't enough hours in the day and I'm not compensated for work on this project as it's fully free/open-source). I encourage those who run into issues getting set up and then get it figured out to submit a pull request (PR) to the readme/add additional documentation to help others who might be confused down the road. I will gladly review/merge PRs that help make things clearer, I am well aware that this is a very confusing plug-in! :)
